sql-plugin/src/main/java/com/nvidia/spark/rapids/SlicedGpuCompressedColumnVector.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids;
+
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.sql.vectorized.ColumnVector;
+import org.apache.spark.sql.vectorized.ColumnarArray;
+import org.apache.spark.sql.vectorized.ColumnarMap;
+import org.apache.spark.unsafe.types.UTF8String;
+
+/**
+ * Wraps a GpuColumnVector but only points to a slice of it.  This is intended to only be used
+ * during shuffle after the data is partitioned and before it is serialized.
+ */
+public class SlicedGpuCompressedColumnVector extends ColumnVector {
```

Comment:
nit: I would do something specific to this. Just because a GpuCompressedColumnVector is not really what this is holding.